Defining The Hypothesis

H0: There is no association between the type of smart home device purchased and the customer satisfaction level.

H1: There is an association between the type of smart home device purchased and the customer satisfaction level.

In [ ]:
from scipy.stats import chi2_contingency
from matplotlib import pyplot as plt
import pandas as pd
from scipy import stats

In [ ]:
data = [[50,70],
        [80,100],
        [60,90],
        [30,50],
        [20,50]]
data

[[50, 70], [80, 100], [60, 90], [30, 50], [20, 50]]

In [ ]:
df=pd.DataFrame(data,
                index=['Very Satisfied','Satisfied','Neutral','Unsatisfied','Very Unsatisfied'],  # row names
                columns=['Very Unsatisfied','Smart Light'])  # column names

In [ ]:
df

Very Unsatisfied  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

Computing Chi-square Statistic

In [ ]:
chi2,p,dof,exp=chi2_contingency(df)

In [ ]:
chi2_contingency(df)

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

Determining the values

In [ ]:
# chi-square statistis
chi2

5.638227513227513

In [ ]:
# p value
p

0.22784371130697179

In [ ]:
# Degree of Freedom
dof

4

In [ ]:
# expected frequency
exp

array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]])

In [ ]:
# alpha value
a = 0.05

In [ ]:
# critical value
critical_value = stats.chi2.ppf(1 - a, dof)
critical_value

9.487729036781154

Conclusion

Since the Chi-Square statistic (5.638) is less than the critical value (9.487), and the p-value (0.227) is greater than the significance level (0.05), so we do not reject the null hypothesis.

Hence we can say that,

There is no significant association between the type of smart home device purchased and the customer satisfaction level.